src/http/server/server.cr

```diff
@@ -127,14 +129,31 @@ class HTTP::Server
   def initialize(@host, @port, @handler)
   end
 
+  def port
+    if server = @server
+      server.local_address.port.to_i
+    else
+      @port
+    end
+  end
+
+  def bind
+    return if @server
+    @server = TCPServer.new(@host, @port)
+  end
```

Comment:
Maybe use `@server ||= TCPServer.new(@host, @port)` so the return type is `TCPServer` instead of `TCPServer|Nil` ?